lib/migration.pm

```diff
@@ -88,5 +89,11 @@ sub disable_installation_repos {
     }
 }
 
+# Record disk info to help debug diskspace exhausted
+# issue during upgrade
+sub record_disk_info {
+    assert_script_run "(which btrfs 2>/dev/null && btrfs filesystem show; which snapper 2>/dev/null && snapper list; df -h) &> /tmp/df.txt";
```

Comment:
That would fail if the fs is not btrfs. We already record the df output in install_and_reboot. I recommend to use the same approach

Reply:
Yes, I find the related code in lib/y2logsstep.pm function post_fail_hook.
Thanks.

Reply:
Updated:
   * separate btrfs logs
   * also record installed rpms

Verified with test run:
http://10.67.18.143/tests/173#step/patch_before_migration/87